lib/discovery.js

```diff
@@ -341,7 +341,7 @@ function mixinDiscovery(MsSQL) {
 
   MsSQL.prototype.buildPropertyType = function(columnDefinition, dataLength) {
     var mysqlType = columnDefinition.dataType;
-    var dataLength = columnDefinition.dataLength;
+    dataLength = dataLength || columnDefinition.dataLength;
```

Comment:
@raymondfeng there is a bug in this function's input:
according to refactored `buildPropertyType` in [loopback-connector](https://github.com/strongloop/loopback-connector/blob/5307f6836492d9385842f46339796cf5cfe280c2/lib/sql.js#L1878), the second input here should be `options` instead of `dataLength`.

So here the logic would be 
```js
MsSQL.prototype.buildPropertyType = function(columnDefinition, options) {
// ...
var dataLength = options.dataLength || columnDefinition.dataLength;
// ...
}
```